Refactoring of saveFormData method validations

denduluriramasaimohanakhil opened the pull request:
Refactoring of saveFormData method validations

Mohanakhil committed:
Testcases

dev:
 coverage:55%
 lint:175
TestCases:
 coverage:48%
 lint:175

 Assertions are corrected

Mohanakhil committed:
Testcases

dev:
 coverage:55%
 lint:175
TestCases:
 coverage:47%
 lint:175

 Assertions are corrected

VaibhavTechsophy commented on src/main/java/com/techsophy/tsf/runtime/form/dto/Component.java:
```diff
@@ -0,0 +1,137 @@
+//
+//    }
+
+    public Map<String,Object> getData(Map<String,Object> parentData,String type,String label)
```
what are type,label parameters for?

VaibhavTechsophy commented on src/main/java/com/techsophy/tsf/runtime/form/dto/Component.java:
```diff
@@ -0,0 +1,137 @@
+    public Map<String,Object> getData(Map<String,Object> parentData,String type,String label)
+    {
+            if(this.type.equals("container")||this.type.equals("datamap"))
+            {
```
all three conditions are returning the same thing except tree, reduce them into one as an else

VaibhavTechsophy commented on src/main/java/com/techsophy/tsf/runtime/form/dto/Component.java:
```diff
@@ -0,0 +1,137 @@
+
+    public boolean isContainer(Component component)
+    {
+        if(component.getComponents()!=null||component.getColumns()!=null||
```
this can directly be the return statement

VaibhavTechsophy commented on src/main/java/com/techsophy/tsf/runtime/form/dto/Validate.java:
```diff
@@ -0,0 +1,130 @@
+   String integer;
+
+
+   public List<ValidationResult> validate(ComponentData compData,GlobalMessageSource globalMessageSource,MongoTemplate mongoTemplate)
```
why we need gloabal message source here, remove it.

VaibhavTechsophy replied:
handle that in the execption class or someplace else

VaibhavTechsophy commented on src/main/java/com/techsophy/tsf/runtime/form/dto/Validate.java:
```diff
@@ -0,0 +1,130 @@
+      String value = String.valueOf(dataMap.get(label)==null?"":dataMap.get(label));
+      String prefix=compData.prefix;
+      String formId=compData.formId;
+      if(component.getValidate().required&&(!dataMap.containsKey(label)))
```
why we need to refer to component.getValidate here, the validation must be for this validate object

VaibhavTechsophy commented on src/main/java/com/techsophy/tsf/runtime/form/dto/Validate.java:
```diff
@@ -0,0 +1,130 @@
+   public List<ValidationResult> validate(ComponentData compData,GlobalMessageSource globalMessageSource,MongoTemplate mongoTemplate)
+   {
+      List<ValidationResult> validationResultList=new ArrayList<>();
+      Component component=compData.comp;
```
make component and data private, and only expose helper methods to retrieve specific details.

VaibhavTechsophy commented on src/main/java/com/techsophy/tsf/runtime/form/dto/Validate.java:
```diff
@@ -0,0 +1,130 @@
+      {
+         if(Double.parseDouble(value)<component.getValidate().getMin())
+         {
+            validationResultList.add(new ValidationResult(label,FORM_DATA_MIN_VALUE_CONDITION_FAILED,globalMessageSource.get(FORM_DATA_MIN_VALUE_CONDITION_FAILED,label)).addPrefix(prefix));
```
why is this dealing with prefix, it should only tell whether the component is valid or not.

VaibhavTechsophy commented on src/main/java/com/techsophy/tsf/runtime/form/dto/Validate.java:
```diff
@@ -0,0 +1,130 @@
+              validationResultList.add(new ValidationResult(label,FORM_DATA_REGEX_CONDITION_FAILED,globalMessageSource.get(FORM_DATA_REGEX_CONDITION_FAILED,label)).addPrefix(prefix));
+          }
+      }
+      if(component.getValidate().getUnique())
```
such a validation should not be done here.

VaibhavTechsophy commented on src/main/java/com/techsophy/tsf/runtime/form/service/impl/FormValidationServiceImpl.java:
```diff
@@ -0,0 +1,162 @@
+            }
+            if(x.get("type").equals("tree"))
+            {
+                List<LinkedHashMap> componentList= (List<LinkedHashMap>) x.get("components");
```
why is this mutation required cannot you manage it with getData from component.

VaibhavTechsophy commented on src/main/java/com/techsophy/tsf/runtime/form/service/impl/FormValidationServiceImpl.java:
```diff
@@ -0,0 +1,162 @@
+        return validationResultList;
+    }
+
+    public void internalLoop(String prefix,Component component,List<ValidationResult> validationResultList,List<Component> componentList,Map<String,Object> data,String formId)
```
this should not eb a public method

VaibhavTechsophy commented on src/main/java/com/techsophy/tsf/runtime/form/service/impl/FormValidationServiceImpl.java:
```diff
@@ -0,0 +1,162 @@
+        List<ValidationResult> validationResultList=new ArrayList<>();
+        if(component.isContainer(component)&&!component.getType().equals("address"))
+        {
+            if(!component.getType().equals("fieldset")&&!component.getType().equals("well")
```
if you need to refer to same value again and again, there is something called variable
string componentType = component.getType()

VaibhavTechsophy commented on src/main/java/com/techsophy/tsf/runtime/form/service/impl/FormValidationServiceImpl.java:
```diff
@@ -0,0 +1,162 @@
+                if(component.getType().equals("tabs"))
+                {
+                    List<Component> componentList=component.getComponents().get(0).getComponents();
+                    internalLoop(prefix,component,validationResultList,componentList,data,formId);
```
why cannot you just loop here itself?

VaibhavTechsophy commented on src/main/java/com/techsophy/tsf/runtime/form/service/impl/FormValidationServiceImpl.java:
```diff
@@ -0,0 +1,162 @@
+        {
+            if(component.getValidate()!=null)
+            {
+                return component.getValidate().validate(new Validate.ComponentData(component,data,prefix,formId),globalMessageSource,mongoTemplate);
```
if you are not able to follow the approach discuss with me, this whole class needs to be refactored

VaibhavTechsophy reviewed (changes requested):
We need to sit  and refactor major portions.

Mohanakhil committed:
Testcases

dev:
 coverage:55%
 lint:175
TestCases:
 coverage:48%
 lint:175

 Assertions are corrected

Mohanakhil committed:
Testcases

dev:
 coverage:55%
 lint:175
TestCases:
 coverage:43%
 lint:175

 Assertions are corrected

Mohanakhil committed:
Testcases

dev:
 coverage:51%
 lint:175
TestCases:
 coverage:51%
 lint:170

 Assertions are corrected

VaibhavTechsophy reviewed (approved):
Good Effort.
Please deploy to test environment so that Mansa could test.
